### PR TITLE
fix rotation matrix;

### PR DESCRIPTION
**Pull request by schmidtijoe**
suggested fix for rotation matrix calculation

### CODE REVIEW

**Comment by pehses** [comment]
I'm not really familiar with this feature (or geometry.py in general).

@schaten can you have a quick look?

File: twixtools/geometry.py [hunk 1]
```diff
@@ -98,30 +98,39 @@ def from_twix(self, twix):
...
-            mat = np.eye(3) * c @ mat
+            # we have cosine 1 or -1, two vectors are (anti-) parallel
+            mat = np.matmul(np.eye(3) * c, init_mat)
         else:
-            V = np.array([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])
-            mat = (np.eye(3) + V + V * V * (1 - c) / s ** 2) @ mat
+            # calculate cross product matrix
+            v_x = np.cross(np.eye(3), v)
+            # calculate rotation matrix, division should be possible from excluding c = -1 above
+            mat = np.eye(3) + v_x + np.divide(np.matmul(v_x, v_x), 1 + c)
```
Comment (schaten) [inline twixtools/geometry.py:131]: Don't get this line - I think it should be mat = np.eye(3) - s * v_x + np.matmul(v_x, v_x) * (1 - c) (then it's [Rodrigues' rotation formula](https://en.wikipedia.org/wiki/Rodrigues%27_rotation_formula#Matrix_notation) ?

**Comment by schaten** [comment]
Thanks @schmidtijoe ! The existing code was definitely wrong, probably my fault.

However I don't get that last line in your pull request. Can you comment on that?

**Comment by schmidtijoe** [comment]
Hi @schaten,
Thanks for looking into this. I guess the difference is: Rodrigues formula gives the rotation matrix to rotate a vector v around the axis k (normalized) about angle phi. What i tried to give is the rotation matrix for the rotation of a vector v to align with the another vector v2 (both normalized).

The jist is (with abbreviated sine and cosine):
To observe that v1 x v2  is sin(angle) * n of a unit vector n, since both vectors are normalized. Hence if we replace K above in rodrigues formula with sin(angle) K and rearrange (using s^2 + c^2 = 1  -->  (1 - c) = s^2 / (1 + c) we end up with the formula given in my code suggestion.
To arrive at the same with the classical version of Rodrigues formula (you have posted) i guess the difference really is only to normalize the calculated cross product of v1 x v2 (in above code this would mean normalizing v before putting it into the cross product matrix v_x), since this is the normalized rotation axis in Rodrigues formula.

**Comment by pehses** [comment]
Great, thanks both of you!

**Pull request merged**